spark3/src/main/java/org/apache/iceberg/spark/Spark3Util.java

```diff
@@ -543,4 +550,63 @@ private static String sqlString(org.apache.iceberg.expressions.Literal<?> lit) {
       }
     }
   }
+
+  public static CatalogAndIdentifier catalogAndIdentifier(SparkSession spark, String name) {
+    try {
+      return catalogAndIdentifier(spark, spark.sessionState().sqlParser().parseMultipartIdentifier(name));
+    } catch (ParseException e) {
+      throw new IllegalArgumentException(String.format("Cannot parse identifier %s", name), e);
+    }
+  }
+
+  /**
+   * A modified version of Spark's LookupCatalog.CatalogAndIdentifier.unapply
+   * Attempts to find the catalog and identifier a multipart identifier represents
+   * @param spark Spark session to use for resolution
+   * @param nameParts Multipart identifier representing a table
+   * @return The CatalogPlugin and Identifier for the table
+   */
+  public static CatalogAndIdentifier catalogAndIdentifier(SparkSession spark, Seq<String> nameParts) {
```

Comment:
Can we avoid Scala in this API to avoid breaking changes? I think converting to `List` immediately and then calling this method is going to be better.

Reply:
Already upset about that Scala 2.13 change :) Yeah no problem, I'll stay Java only in message signatures with the exception of Spark Specific classes.